include/mujincontrollerclient/mujinjson.h

```diff
@@ -501,6 +502,26 @@ template<class U> inline void LoadJsonValue(const rapidjson::Value& v, std::map<
     }
 }
 
+template<class U> inline void LoadJsonValue(const rapidjson::Value& v, std::unordered_map<std::string, U>& t) {
+    // It doesn't make sense to construct an unordered map from anything other
+    // than a full JSON object
+    if (!v.IsObject()) {
+        throw MujinJSONException("Cannot convert json type " + GetJsonTypeName(v) + " to unordered_map");
+    }
+
+    // Ensure our output is a blank slate
+    t.clear();
+    U value;
+    for (rapidjson::Value::ConstMemberIterator it = v.MemberBegin();
+         it != v.MemberEnd(); ++it) {
+        LoadJsonValue(it->value, value);
+        // Our key name needs to be explicitly length-constructed since it may
+        // contain \0 bytes.
+        // Move our temporary value to try and minimize copying of complex types
+        t[std::string(it->name.GetString(), it->name.GetStringLength())] = std::move(value);
```

Comment:
> std::move(value)

interesting. can we add similar optimization to other containers?

Reply:
(maybe just LoadJsonValue(map)?)

Reply:
Yeah just removing the temporary altogether would avoid the issue, since you can then construct in-place.

Reply:
Could you `LoadJsonValue(it->value, t[std::string(it->name.GetString(), it->name.GetStringLength()]);`?

Reply:
Yep, just pushed that change in fact.

Reply:
Changed it for ordered `map` as well. Vector and array already avoid extra copies.